xen/drivers/passthrough/iommu.c

```diff
@@ -129,7 +129,7 @@ static void __init parse_iommu_param(char *s)
     } while ( ss );
 }
 
-int iommu_domain_init(struct domain *d)
+int iommu_domain_init(struct domain *d, bool_t use_iommu)
```

Comment:
Well at this moment I think of moving these changes to arch_iommu_domain_init(). You need them for ARM only. As I know on AMD(x86) IOMMU does not share pt with the cpu but works well without your changes.